earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -355,5 +362,35 @@ bool StorageManager<AssetType>::SaveDirtyToDotNew(
   return true;
 }
 
+template<class AssetType>
+void StorageManager<AssetType>::WriteDirty() {
+  if (writeDirty) {
+    std::lock_guard<std::recursive_mutex> lock(storageMutex);
+    khFilesTransaction savetrans(".new");
+    typename std::map<AssetKey, PointerType>::iterator entry = dirtyMap.begin();
+    notify(NFY_WARN, "WriteDirty: %s", ToString(typeid(AssetType).name()).c_str());
+    while (entry != dirtyMap.end()) {
+      if (entry->second.use_count() == 2) {
```

Comment:
Eventually we really ought to call WriteDirtyToDotNew here rather than duplicating most of the code.